src/bitmap/bitmap_8k.rs

```diff
@@ -0,0 +1,314 @@
+use crate::bitmap::sorted_u16_vec::SortedU16Vec;
+use crate::bitmap::store::Store;
+use std::borrow::Borrow;
+use std::ops::{BitAndAssign, BitOrAssign, BitXorAssign, RangeInclusive, SubAssign};
+
+pub const BITMAP_LENGTH: usize = 1024;
+
+#[derive(Clone)]
+pub struct Bitmap8K {
+    len: u64,
+    bits: Box<[u64; BITMAP_LENGTH]>,
+}
+
+impl Bitmap8K {
+    pub fn new(len: u64, bits: Box<[u64; BITMAP_LENGTH]>) -> Bitmap8K {
+        Bitmap8K { len, bits }
+    }
```

Comment:
Don't we want this method to be `unsafe` to be sure that no rules are broken, and be sure that the user of this method is careful!

Reply:
From my understanding `unsafe` in rust refers specifically to memory safety, not for anything that could violate some invariant.

I take your point though. Lets make this `pub(crate)`

Reply:
> From my understanding `unsafe` in rust refers specifically to memory safety, not for anything that could violate some invariant.

I am not sure about that, invariants [can be of two kinds: safety and validity](https://internals.rust-lang.org/t/two-kinds-of-invariants-safety-and-validity/8264). In this case we are talking about validity. What if the len doesn't match what's in the Box (got it?)?

Reply:
As I understand this more, we should declare a library function as `unsafe` as soon as it is possible to create an invalid state with the parameters, leading to possible future bugs. It can be called library undefined behavior.

Reply:
We should not make it unsafe, but at list rename it `new_uncheked` and make it private, what do you think?

Reply:
In fact, you don't need to make private `pub(crate)` any method on the containers, those are not accessible.

Reply:
Honestly I'd like to mark all `Container` `Store` `Bitmap8k` `SortedU16Vec` and so on as `pub(crate)` as end users do not interact with these types

Perhaps we can audit the types with a follow up PR before the next breaking point release.

Reply:
`Container`, `Store`, `Bitmap8k`, `SortedU16Vec` are not exposed to the end-users, their modules are not `pub` in lib.rs.

Reply:
> `Container`, `Store`, `Bitmap8k`, `SortedU16Vec` are not exposed to the end-users, their modules are not `pub` in lib.rs.

Ah, OK. That makes sense.

>  We should not make it unsafe, but at list rename it new_unchecked and make it private, what do you think?

It cant be private anyhow, it's required for deserializing.